tests/cases/compiler/scopeCheckStaticInitializer.ts

```diff
@@ -0,0 +1,14 @@
+class X {
```

Comment:
`this.b` is `undefined` in the expression `"" + this.b`, even though its initial value is supposed to be `""`. I think we want to error in this case.